bitsofgood/pencils-for-success
# Endpoint to update Chapter

### Description
Added endpoint `/api/chapters/:id` to update the given chapter. It only allows admin and chapter user for that chapter to update the following fields:
- name
- email

**Related Issues/PRs:** 

- Closes #40

### Test Plan

Test for following unauthorized scenarios:
- Update a chapter without authenticating. It should return a 401 error.
- Update a chapter authenticating as a chapter user but not part of that chapter. It should return a 401 error.

Test for following success scenarios:
- Update a chapter authenticated as an admin
- Update a chapter authenticated as a chapter user for that chapter

Test for unique constraint error:
- Update a chapter to have a duplicate email as another chapter. It should return an error and not update the chapter.

## Notes
We need to add endpoints to update fields for chapter user.
